robot/client/client.go

```diff
@@ -99,6 +165,17 @@ func New(ctx context.Context, address string, logger golog.Logger, opts ...Robot
 		resourceClients:         make(map[resource.Name]interface{}),
 		remoteNameMap:           make(map[resource.Name]resource.Name),
 	}
+
+	rc.dialOptions = append(
+		rc.dialOptions,
+		// error handling
+		rpc.WithUnaryClientInterceptor(rc.handleUnaryDisconnect),
+		rpc.WithStreamClientInterceptor(rc.handleStreamDisconnect),
+		// operations
+		rpc.WithUnaryClientInterceptor(operation.UnaryClientInterceptor),
+		rpc.WithStreamClientInterceptor(operation.StreamClientInterceptor),
+	)
+
```

Comment:
interceptors are applied in order from first to last

Reply:
maybe just add that as a nice little helpful comment